### PR TITLE
Update Gazebo7 linking method

### PR DESCRIPTION
**Pull request by davetcoleman**
CMake tweaks for Kinetic

*This PR is sponsored by [Vicarious](http://www.vicarious.com/)*

### CODE REVIEW

File: baxter_gazebo/CMakeLists.txt [hunk 1]
```diff
@@ -1,31 +1,27 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(baxter_gazebo)
 
-SET(CMAKE_CXX_FLAGS "-std=c++11")
```
Comment (davetcoleman) [inline baxter_gazebo/CMakeLists.txt:4]: This is automatically set by the line ``+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GAZEBO_CXX_FLAGS}")``

File: baxter_gazebo/CMakeLists.txt [hunk 2]
```diff
@@ -1,31 +1,27 @@
...
+    roscpp
+    gazebo_ros
+    gazebo_ros_control
+    controller_manager
+    baxter_core_msgs
 )
 
-find_package(Boost REQUIRED COMPONENTS system)
+# Depend on system install of Gazebo
+find_package(GAZEBO REQUIRED)
```
Comment (davetcoleman) [inline baxter_gazebo/CMakeLists.txt:14]: Based on @j-rivero's PR https://github.com/ros-simulation/gazebo_ros_pkgs/pull/521/files#diff-bfb95e6fc582baf511a43d0fa5f77854R41

File: baxter_gazebo/CMakeLists.txt (hunk 1 above)
Comment (IanTheEngineer) [reply baxter_gazebo/CMakeLists.txt:4]: +1

File: baxter_gazebo/CMakeLists.txt (hunk 2 above)
Comment (IanTheEngineer) [reply baxter_gazebo/CMakeLists.txt:14]: Ah, sounds good.

**Review by IanTheEngineer — APPROVED** [review]

**Pull request merged**